application-xpoll-ui/src/main/resources/XPoll/XPollMigration.xml

```diff
@@ -0,0 +1,175 @@
+<?xml version="1.1" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.3" reference="XPoll.XPollMigration" locale="">
+  <web>XPoll</web>
+  <name>XPollMigration</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>Main.WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title>XPollMigration</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity}}
+#if ($hasAdmin &amp;&amp; $request.migrate &amp;&amp; $services.csrf.isTokenValid($request.form_token))
+  #set ($xpollPages = $services.query.xwql('where doc.object(XPoll.XPollClass).type in (:types)').bindValue('types', ['0', '1']).execute())
+  #set ($types = {
+    '0': 'multi',
+    '1': 'single'
+  })
+  #foreach ($item in $xpollPages)
+    #set ($document = $xwiki.getDocument($item))
+    #set ($xpollObj = $document.getObject('XPoll.XPollClass'))
+    #set ($pollType = $xpollObj.getValue('type'))
+    #set ($discard = $document.use($pollObj))
+    #set ($discard = $document.set('type', $types.get($pollType)))
```

Comment:
```suggestion
    #set ($discard = $xpollObj.set('type', $types.get($pollType)))
```

Since you have the object, you could set the property value directly using it.